python/lsst/daf/butler/core/timespan.py

```diff
@@ -749,6 +749,30 @@ def contains(self: _S, other: Union[_S, sqlalchemy.sql.ColumnElement]) -> sqlalc
         """
         raise NotImplementedError()
 
+    @abstractmethod
+    def lower(self: _S) -> sqlalchemy.sql.ColumnElement:
+        """Return a SQLAlchemy expression representing a lower bound of a
+        timespan.
+
+        Returns
+        -------
+        lower : `sqlalchemy.sql.ColumnElement`
+            A SQLAlchemy expression for a lower bound.
+        """
+        raise NotImplementedError()
```

Comment:
If the tore value is NULL then NULL should be returned, the issue is of course that NULL can be handled differently (or unexpectedly) in ORDER BY. If we could agree on something specific that should be returned for NULL then we can wrap it into `COALESCE` to return a value instead of NULL. Should NULL be treated as EMPTY? Or maybe they should be ordered earlier than any non-NULL timespans, so that they don't appear in a query "give me the latest timestamp".